### ISSUE TITLE
Complex numbers are not numbers - a debate

### ISSUE THREAD

**Issue by endremborza** [leios/SoME_Topics#164]
## About the author

I am Endre, a data scientist in academia - sort of a research software engineer. Most of my current work deals with statistics, programming, complexity and scaling but I think I might apply to cs phd programs with a little more theory.  Also, I have a problem with complex numbers.

## Quick Summary

I think you can convey interesting ideas about the generalization of arithmetic operations, hyperoperations, ordered fields, and the usefulness of abstractions by having/presenting a debate about the status of complex numbers.

In technical terms, complex numbers are algebraically closed, but real numbers and their subsets are not, while complex numbers cannot be an ordered field but real numbers and their subsets can. I think you can build up to the contrasting statements "In order to be *numbers* a field needs to be ordered" and "If a field is algebraically closed, it is definitely *numbers*" in an educational and entertaining way. This is what I would like to try. Get maybe a little philosophical, but mainly get a message through about the purpose of math and numbers.

## Target medium

I would like to frame this as a debate. A video confronting the two sides of the argument, either in a witty, insulting, epic rap battle style or more reserved and academic. Hopefully with some drawings and/or animation. I am willing and prepared to work on this myself, but finding someone to collaborate with sounded like fun. Plus, I am open to ideas and feedback, so I thought I'd share.

### More details

Here is a *very* rough draft of the conversation, **C** for complex numbers, **R** against:

- **C**: Numbers are meant to be manipulated, and operations should be easily performed on them: addition, subtraction, multiplication, division... This is why the Hindu-Arabic numeral system is great and the Roman is stupid.
- **R**: That's way too complicated. The purpose of numbers is comparison. As long as you can tell which number is larger, you are just fine. The Romans didn't do that badly.
-  **C**: Yes, but you can close the circle. It is so logical it hurts:
    - the number one + counting -> natural numbers
    - repeated counting = addition
    - inverse of addition = subtraction
    - natural numbers + subtraction -> integers
    - repeated addition = multiplication
    - inverse of multiplication = division
    - integers + division -> rational numbers
    - repeated multiplication = exponentiation
    - inverse of exponentiation = nth root extraction and logarithm
    - at this point, you must use complex numbers, because the logical chain of these operations demands it. even better, after that, you are done.
- **R**: You were done before, you can't properly compare two complex numbers
- **C**: Well, you can do  lexicographical ordering...
- **R**: Yes, and you do the "repeated repeated counting" for two numbers larger than zero and you get a number smaller than zero.
- **C**: You can still compare them. You just have to overcome a little difficulty. A small price to pay for getting all the operations on all the numbers.
- **R**: $\frac{1}{0}$ - *mic drop*

Additional ideas to possibly expand on:
- you can fit all kinds of numbers on a line, you cannot fit complex numbers meaningfully on a line, even though they have the same cardinality as real numbers
- when $\sqrt{-1} $ didn't work, we made up *i* for it, why can we not make something up for $\frac{1}{0}$ and call it a day?

### Contact details

Probably best as a reply to this issue, but you can reach out at [endremborza@gmail.com](mailto:endremborza@gmail.com)

**Comment by robertefry** [comment]
Technically, you can divide by zero. Just not under the real number system.

Enter [Wheel Theory](https://www.youtube.com/watch?v=ydLTfyXaQmU). Like 0 is an absorbing element of the real numbers, we tack on a null element to the number system which is also absorbing (like infinity). The caveat to this is all our standard definitions for absorbtion need to be changed.

Having done this, we see division by zero is the null element.

An example of this is the projective plane - the entire real numberline is folded into a wheel such that we join +∞ = -∞

**Comment by endremborza** [comment]
Thanks. That's a nice extension for the zero division problem, and might be - or probably should be - worked in a little differently than I originally thought.

**Comment by SimonAndreys** [comment]
I would like to see this video, but taken to its extreme : R wins against C, but enters Q with other persuasive arguments to why real numbers are not actually numbers (you can't actually write down an irrational number in finite time, besides we don't even know whether R's cardinal is aleph_1 or higher !).  R is defeated but enters Z (Q is just an artificial construction of equivalence of couples, the actual numbers are actually Z), N eliminates Z because negative numbers are to abstract, N* enters with the argument that zero is not actually a number (how can you count something which is not there), finally {1} arrives : all other numbers are made of one's. In the distance, the emptyset stares at the others with scorn.

**Comment by filonik** [comment]
Yeah, sounds like a fun video. Also, I like where @[SimonAndreys](https://github.com/SimonAndreys) is going. In the way R is defined as a completion of Q with respect to a metric (i.e. how you compare them), you can chose a different metric and end up with a different set of numbers ([see Wikipedia](https://en.wikipedia.org/wiki/Construction_of_the_real_numbers#Construction_from_Cauchy_sequences)):

> R is defined as the completion of Q with respect to the metric |x-y|, as will be detailed below (for completions of Q with respect to other metrics, see p-adic numbers.

I.e. this could be another argument in the debate calling into question R's special status.

**Comment by Nikolaj-K** [comment]
>I think you can build up to the contrasting statements "In order to be numbers a field needs to be ordered" and "If a field is algebraically closed, it is definitely numbers" in an educational and entertaining way.

Have you heard someone actually taking the latter position? I don't think people will defend that some power series are numbers.

**Comment by endremborza** [comment]
Well, you sent me down a rather interesting rabbit hole from power series to p-adic numbers (also an interesting candidate for *number* )

Nevertheless yes, I suppose this way it is quite a bit too general. Let me try to rephrase it to "Being the algebraic closure of real numbers qualifies the complex numbers to be called numbers". I can imagine taking the position that "The whole point of creating numbers was to arrive at an algebraically closed field"

**Comment by kevinb9n** [comment]
The debate format sounds fun. Someone contact Lin-Manuel Miranda!

**Comment by Dayn-Hues** [comment]
I think this is a fun idea. I am a high school teacher and I always really like having conversations with students about why we actually use imaginary numbers and about how they got named that and everything. Just a quick thing, if you haven't already. I would highly recommend checking out [Imaginary Numbers are Real](https://www.youtube.com/watch?v=T647CGsuOVU&list=PLiaHhY2iBX9g6KIvZ_703G3KJXapKkNaF) by Welch Labs. Long videos but really well made series about the history and rationale behind imaginary numbers. This may be totally different from what you are thinking but I think if you are going to do anything about complex numbers this series is definitely worth checking out. I also really like @SimonAndreys take.

**Comment by alan2here** [comment]
You don't even need to start with "1" to do repeated addition, you can start with simply "a value you know nothing about" and a function that always gives you a new thing whenever it's given a thing, but will always give the same output for a given input:

for example
A --> Y
Y --> $
$ --> !
etc…

You may need limits as well if you want all irrationals, which even includes non-computable and eventually non-describable reals.

If you'd like to build higher while keeping things both Well Ordered and a Field then you can build all the way up to the surreals, you could optionally specify finite and describable as well if you like.

**Comment by fbunc** [comment]
I've been thinking about complex numbers representing information in the context of time series and data-science:

https://github.com/fbunc/ROOTS_OF_UNITY_AND_ROULETTE_AS_A_BIASED_DICE/blob/master/00_ROOTS_OF_UNITY_AND_GENERATING_FUNCTIONS.ipynb

Maybe we can collaborate

**Comment by tariq3erwa** [comment]
I think this debate could be used to introduce the audience to geometric (or clifford) algebras, in this mathematical frame there is a general definition of a complex number as a subset of multivectors.

**Comment by fbunc** [comment]
https://github.com/fbunc/golden_ratio_mod_13_primes_disc/blob/master/Eigenfunction_mod_13_golden_wheel.ipynb

https://www.youtube.com/watch?v=NY5tgKBtLQ0
